charts/openshift-metering/templates/openshift-reporting/report-queries/pod-memory.yaml

```diff
@@ -371,11 +371,11 @@ spec:
       usage.pod_usage_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_usage_percent,
       request.pod_request_memory_byte_seconds / capacity.total_cluster_capacity_memory_byte_seconds AS pod_memory_request_percent,
       capacity.total_cluster_capacity_memory_byte_seconds
-    FROM namespace_memory_request as request
-    JOIN namespace_memory_usage as usage
-      ON request.period_start = usage.period_start
-      AND request.period_end = usage.period_end
-      AND request.namespace = usage.namespace
+    FROM namespace_memory_usage as usage
+    JOIN namespace_memory_request as request
+      ON usage.period_start = request.period_start
+      AND usage.period_end = request.period_end
+      AND usage.namespace = request.namespace
```

Comment:
I don't think swapping the tables in the from and join clauses will accomplish much as the default join technique is an inner join, which we're seeing is problematic in the case where the namespace_memory_request table has less row results than the namespace_memory_usage table. I think we should just update to use a right outer join instead.

@bentito any thoughts on this approach? For context, we were missing results for the namespace memory utilization report, where Pods in a namespace were using memory, but weren't being listed. When I dug into this a bit more, this was because the Pods listed in that missing namespace weren't setting memory requests, which results in a row mismatch between the usage and request tables. The result is that we attempt to join any matching results in both of those tables, but we neglect the case where a Pod in a namespace uses memory, but doesn't set a request, which doesn't seem like the correct behavior.

Reply:
yep, changing the join type to favor the thing that definitely generates rows seems like the right fix here.